tic_tac_toe/game.py

```diff
@@ -87,18 +149,52 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    board = game['board']
+    current_player = get_next_turn(game)
+
+    if _board_is_full(board) or game['winner'] != None:
+        raise InvalidMovement('Game is over.')
+
+    if current_player != player:
+        raise InvalidMovement('"' + current_player + '" moves next.')
+
+    if not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+
+    if not _position_is_empty_in_board(position, board):
+        raise InvalidMovement('Position already taken.')
+
+    game['board'][position[0]][position[1]] = player
+
+    winner = _check_winning_combinations(board, player)
+
+    if winner:
+        game['winner'] = player
+        raise GameOver('"' + player + '" wins!')
+    elif _board_is_full(board):
+        raise GameOver('Game is tied!')
+    else:
+        _switch_turns(game, player)
```

Comment:
thanks! it was an `if/else` block at first but I refactored it after we got all the tests to pass on travis. I prefer abstracting away stuff wherever possible because it's easier to modify later on.